test/code/omi_preexec/config_scanner_test.cpp

```diff
@@ -0,0 +1,231 @@
+ * SOFTWARE.
+ *
+ * ---------------------------------- END OF LICENSE -----------------------------
+*/
+/*--------------------------------------------------------------------------------
+
+    Created date    2016-06-09 08:31:00
+
+    MySQL Config file loading unit tests.
+
+*/
+/*----------------------------------------------------------------------------*/
+
+#include <scxcorelib/scxcmn.h>
+#include <scxcorelib/scxexception.h>
+#include <scxcorelib/scxfile.h>
+#include <scxcorelib/stringaid.h>
+#include <testutils/scxunit.h>
+#include <testutils/providertestutils.h>
+
+#include <iostream> // for cout
+#include <vector>
+#include <map>
+
+#include "mycnf-scanner.h"
+
+using namespace SCXCoreLib;
+
+class MySQL_ScannerTestAuthenticationDependencies : public MySQL_AuthenticationDependencies
+{
+public:
+    virtual SCXCoreLib::SCXFilePath GetDefaultAuthFileName(uid_t /* uid */) const
+    {
+        return L"./does-not-exist.cnf";
+    }
+
+    virtual bool Force_AutoUpdate_Flag() { return true; }
+};
+
+class ConfigScanner_Test : public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE( ConfigScanner_Test );
+
+    CPPUNIT_TEST( testCnfScan_RHELCnf_Success );
+    CPPUNIT_TEST( testCnfScan_SLESCnf_Success );
+    CPPUNIT_TEST( testCnfScan_CnfWithIncludeFileDirective_Success );
+    CPPUNIT_TEST( testCnfScan_DebianCnf_Success );
+    CPPUNIT_TEST( testCnfScan_DirectivesFormHierarchy_Success );
+    CPPUNIT_TEST( testCnfScan_CnfWithIncludeFileIncludesItself_Throws );
+    CPPUNIT_TEST( testCnfScan_IncludeDirsFormLoop_Throws );
+    CPPUNIT_TEST( testCnfScan_CnfFileMissingIncludeDir_Throws );
+    CPPUNIT_TEST( testCnfScan_CnfFileMissingInclude_Throws );
+    CPPUNIT_TEST( testCnfScan_InvalidIncludeDir_Throws );
+    CPPUNIT_TEST( testCnfScan_InvalidInclude_Throws );
+
+    CPPUNIT_TEST_SUITE_END();
+
+private:
+
+public:
+    void setUp(void)
+    {
+    }
+
+    void tearDown(void)
+    {
+    }
+
+    void ValidateAuth(std::map< unsigned int, std::string >& expected, MySQL_Authentication& auth)
+    {
+        std::vector<unsigned int> authPorts;
+        auth.GetPortList( authPorts );
+        CPPUNIT_ASSERT_EQUAL(expected.size(), authPorts.size());
+        for (std::vector<unsigned int>::const_iterator it = authPorts.begin(); it != authPorts.end(); ++it)
+        {
+            MySQL_AuthenticationEntry authEntry;
+            auth.GetEntry(*it, authEntry);
+            CPPUNIT_ASSERT_EQUAL(expected[authEntry.port], authEntry.binding);
+        }
+    }
+
+    void testCnfScan_RHELCnf_Success()
+    {
+        MySQL_Authentication auth( SCXHandle<MySQL_AuthenticationDependencies>(new MySQL_ScannerTestAuthenticationDependencies()));
+        auth.Load();
+
+        std::map< unsigned int, std::string > portBindingMap;
+        portBindingMap[3306] = "127.0.0.1";
+
+        MySQL::MySQL_ConfigurationScanner scanner;
+        scanner.Scan(L"cnfs/rhel/my.cnf", auth);
```

Comment:
I really liked how this worked out, by the way. The tests are MUCH smaller and simpler this way, and it's much easier to add new tests. Very nicely done.

---
In reply to: [67190807](https://github.com/Microsoft/MySQL-Provider/pull/5#discussion_r67190807) [](ancestors = 67190807)